instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/TracingRequestStreamWrapper.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awslambda.v1_0;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Wrapper for {@link TracingRequestStreamHandler}. Allows for wrapping a regular lambda, enabling
+ * single span tracing. Main lambda class should be configured as env property OTEL_LAMBDA_HANDLER
+ * in package.ClassName::methodName format. Lambda class must implement {@link
+ * RequestStreamHandler}.
+ */
+public class TracingRequestStreamWrapper extends TracingRequestStreamHandler {
+
+  private WrappedLambda wrappedLambda;
+
+  @Override
+  protected void doHandleRequest(InputStream input, OutputStream output, Context context)
+      throws IOException {
+
+    if (wrappedLambda == null) {
```

Comment:
I wonder if we should just make this static since the configuration is?

Reply:
Good idea, changed.